src/lsm/manifest_log.zig

```diff
@@ -1239,6 +1239,14 @@ const Pace = struct {
         /// The number of live tables.
         tables_count: u32,
     }) u32 {
+        if (options.tables_count > pace.tables_max) {
+            vsr.fatal(
+                .forest_tables_count_would_exceed_limit,
+                "forest_tables_count would exceed limit (tables_count={} tables_max={}) - " ++
+                    "please contact the team directly who will be able to assist",
+                .{ options.tables_count, pace.tables_max },
+            );
+        }
```

Comment:
I'd say a good rule-of-thumb for this sort of checks is to detect the limit just before it is exceeded, rather than at some later point. Perhaps there are _other_ code paths that rely on the fact that table_max is not exceeded?

So, when there's an invariant guarding data D, check it when D is modified, not when it is read. 

Here, I think we should keep this assert as is, and rather add a check at `open_read_block_callback` and `append_internal`?

Reply:
> So, when there's an invariant guarding data D, check it when D is modified, not when it is read.

Agreed!

It was a little tricky here, because in `append_internal` it uses `getOrPutAssumeCapacity`, and it is allowable for a `.insert` to overwrite a previous record due to compaction... So I went with a bit of a creative solution.